test/widgets/emoji_reaction_test.dart

```diff
@@ -284,4 +296,184 @@ void main() {
+
+    Condition<Object?> conditionEmojiListEntry({
+      required ReactionType emojiType,
+      required String emojiCode,
+      required String emojiName,
+    }) {
+      return (Subject<Object?> it) => it.isA<EmojiPickerListEntry>()
+        ..emoji.which((it) => it
+          ..emojiType.equals(emojiType)
+          ..emojiCode.equals(emojiCode)
+          ..emojiName.equals(emojiName));
+    }
+
+    List<Condition<Object?>> arePopularEntries = popularCandidates.map((c) =>
+      conditionEmojiListEntry(
+        emojiType: c.emojiType,
+        emojiCode: c.emojiCode,
+        emojiName: c.emojiName)).toList();
+
+    testWidgets('show, search', (tester) async {
+      final message = eg.streamMessage();
+      await setupEmojiPicker(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+      check(tester.widgetList<EmojiPickerListEntry>(find.byType(EmojiPickerListEntry))).deepEquals([
+        ...arePopularEntries,
+        conditionEmojiListEntry(
+          emojiType: ReactionType.realmEmoji,
+          emojiCode: '1',
+          emojiName: 'buzzing'),
+        conditionEmojiListEntry(
+          emojiType: ReactionType.zulipExtraEmoji,
+          emojiCode: 'zulip',
+          emojiName: 'zulip'),
+        conditionEmojiListEntry(
+          emojiType: ReactionType.unicodeEmoji,
+          emojiCode: '1f4a4',
+          emojiName: 'zzz'),
+      ]);
+
+      await tester.enterText(searchFieldFinder, 'z');
+      await tester.pump();
+
+      check(tester.widgetList<EmojiPickerListEntry>(find.byType(EmojiPickerListEntry))).deepEquals([
+        conditionEmojiListEntry(
+          emojiType: ReactionType.zulipExtraEmoji,
+          emojiCode: 'zulip',
+          emojiName: 'zulip'),
+        conditionEmojiListEntry(
+          emojiType: ReactionType.unicodeEmoji,
+          emojiCode: '1f4a4',
+          emojiName: 'zzz'),
+        conditionEmojiListEntry(
+          emojiType: ReactionType.realmEmoji,
+          emojiCode: '1',
+          emojiName: 'buzzing'),
+      ]);
+
+      await tester.enterText(searchFieldFinder, 'zz');
+      await tester.pump();
+
+      check(tester.widgetList<EmojiPickerListEntry>(find.byType(EmojiPickerListEntry))).deepEquals([
+        conditionEmojiListEntry(
+          emojiType: ReactionType.unicodeEmoji,
+          emojiCode: '1f4a4',
+          emojiName: 'zzz'),
+        conditionEmojiListEntry(
+          emojiType: ReactionType.realmEmoji,
+          emojiCode: '1',
+          emojiName: 'buzzing'),
+      ]);
+
+      debugNetworkImageHttpClientProvider = null;
+    });
+
+    testWidgets('adding success', (tester) async {
+      final message = eg.streamMessage();
+      await setupEmojiPicker(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+      connection.prepare(json: {});
+      await tester.tap(find.descendant(
+        of: find.byType(BottomSheet),
+        matching: find.text('\u{1f4a4}')), warnIfMissed: false); // 'zzz' emoji
```

Comment:
Huh, why warnIfMissed false?
```suggestion
      await tester.tap(find.descendant(
        of: find.byType(BottomSheet),
        matching: find.text('\u{1f4a4}'))); // 'zzz' emoji
```

Seems like if this tap doesn't actually hit, that'd be a bug somewhere.

Reply:
Whoops; this wasn't meant to be included, it's an itermediate debug leftover.
Fixed now.